src/js/components/AdminPanels.js

```diff
@@ -0,0 +1,299 @@
+    (dataStr) => {
+      const data = JSON.parse(dataStr)
+      const updatedMenus = menus.map((menu) => {
+        const value = data[menu.key]
+        const selectedItem = menu.items.find((item) => item.value === value)
+        return { ...menu, selectedItem }
+      })
+      setMenus(updatedMenus)
+    }
+  )
+}
+
+// Custom hook to handle Redis post success/failure
+function useRedisStatus() {
+  const [redisChanged, setRedisChanged] = useState(null)
+
+  const updateRedisStatus = (status) => {
+    setRedisChanged(status)
+    setTimeout(() => setRedisChanged(null), 2000)
+  }
+
+  return [redisChanged, updateRedisStatus]
+}
+
+export default function AdminPanels({
+  initMenus,
+  initAdmin,
+  redisEndpointUrl,
+  redisKeyPrefix,
+  authEndpointUrl,
+}) {
+  const [admin, setAdmin] = useState(initAdmin)
+  const [menus, setMenus] = useState(initMenus)
+
+  const refreshMenus = () =>
+    fetchAndUpdateMenus(redisEndpointUrl, menus, setMenus)
+
+  useEffect(() => {
+    simpleGet(authEndpointUrl)
+      .then((dataStr) => {
+        try {
+          const data = JSON.parse(dataStr)
+          setAdmin((prevAdmin) => ({
+            ...prevAdmin,
+            email: data.email,
+            name: data.name,
+          }))
+        } catch (e) {
+          console.error("Error parsing auth API JSON response:", e)
+        }
+      })
+      .catch((error) => console.warn("Error loading auth API:", error.message))
+  }, [])
+
+  const firstName = admin.name ? admin.name.split(" ")[0] : ""
+
+  return (
+    <ModalProvider>
+      {admin.name && <h3 className="admin-greeting">Hello {firstName}</h3>}
+      <div className="admin-panels-container">
+        <div className="admin-panels-header">
+          <h3 className="admin-panels-title">Redis Controls</h3>
+        </div>
+        <RedisPanel
+          menus={menus}
+          setMenus={setMenus}
+          redisEndpointUrl={redisEndpointUrl}
+          redisKeyPrefix={redisKeyPrefix}
+        />
+        <div className="danger-panel">
+          <div className="danger-panel-border">
+            <h4 className="danger-panel-title">Danger Zone</h4>
+            <AdminDangerPanel
+              refreshMenus={refreshMenus}
+              redisEndpointUrl={redisEndpointUrl}
+            />
+          </div>
+        </div>
+      </div>
+    </ModalProvider>
+  )
+}
+
+AdminPanels.propTypes = {
+  initMenus: PropTypes.array,
+  initAdmin: PropTypes.shape({
+    username: PropTypes.string,
+    email: PropTypes.string,
+    name: PropTypes.string,
+  }),
+  redisEndpointUrl: PropTypes.string,
+  redisKeyPrefix: PropTypes.func,
+  authEndpointUrl: PropTypes.string,
+}
+
+export function RedisPanel({
+  menus,
+  setMenus,
+  redisEndpointUrl,
+  redisKeyPrefix,
+}) {
+  const [refresh, setRefresh] = useState(false)
+
+  useEffect(() => {
+    fetchAndUpdateMenus(redisEndpointUrl, menus, setMenus)
+  }, [refresh, redisEndpointUrl])
+
+  const triggerRefresh = () => setRefresh((prev) => !prev)
```

Comment:
You are having this `refresh` variable only for triggering re-renders right? I think a better approach would be to just pass the `fetchAndUpdateMenus` to the child component. Something like:
```
const handleRefresh = () => {
    fetchAndUpdateMenus(redisEndpointUrl, menus, setMenus)
}
useEffect(() => {
    handleRefresh()
  }, [redisEndpointUrl])
  
<ChildComponent onRefresh={handleRefresh} />
```